TestsFoundation/TestsFoundation/UITests/CoreUITestCase.swift

```diff
@@ -327,9 +330,35 @@ open class CoreUITestCase: XCTestCase {
         send(.setAnimationsEnabled(enabled))
     }
 
-    open func mockNow(_ date: Date) {
-        Clock.mockNow(date)
-        send(.mockNow(date))
+    public enum Permission: String, CaseIterable {
+        case photos = "kTCCServicePhotos"
+        case microphone = "kTCCServiceMicrophone"
+    }
+
+    // Don't rely on this, it won't work on device. It does make simulator tests more reliable
+    open func setSimulatorPermission(_ permission: Permission, allowed: Bool = true) -> Bool {
+        let dbPath = Bundle.main.bundlePath + "/../../../../../Library/TCC/TCC.db"
+        XCTAssert(FileManager.default.fileExists(atPath: dbPath), "couldn't find TCC.db")
+
+        var db: OpaquePointer?
+        let service = permission.rawValue
+        let client = Bundle.main.testTargetBundleID!
+        let query = """
+        delete from access where service = '\(service)' and client = '\(client)';
+        insert into
+        access (service, client, client_type, allowed, prompt_count, csreq, policy_id)
+        values('\(service)', '\(client)', 0, \(allowed ? 1 : 0), 0, 0, 0);
+        """
+        print(query)
+        guard sqlite3_open(dbPath, &db) == SQLITE_OK else { return false }
+        defer { sqlite3_close(db) }
+        var err: UnsafeMutablePointer<Int8>?
+        guard sqlite3_exec(db, query, nil, nil, &err) == SQLITE_OK else {
+            let msg = err.map { String(cString: $0) } ?? "unknown error"
+            print("Error setting permission: \(msg)")
+            return false
+        }
+        return true
```

Comment:
Oh boy! What was the problem before this???

Reply:
The "correct" way of using `addUIInterruptionMonitor` has always been flaky and slow. The nightly for this PR failed on 2 of those tests, so I though I'd fix it more permanently.